Jint/Engine.cs

```diff
@@ -42,6 +42,30 @@ public class Engine
 
         // cache of types used when resolving CLR type names
         internal Dictionary<string, Type> TypeCache = new Dictionary<string, Type>();
+		// cache of learned type conversions.
+		internal Dictionary<Type, Func<Engine, object, JsValue>> TypeMappers = new Dictionary<Type, Func<Engine, object, JsValue>>()
```

Comment:
I'll give this a try, our rules server is already multithreaded, so it would be a good test